x-pack/plugins/observability_solution/apm/public/components/shared/environment_badge/index.tsx

```diff
@@ -7,19 +7,29 @@
 
 import React from 'react';
 import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
 import { ItemsBadge } from '../item_badge';
+import { NotAvailableEnvironment } from '../not_available_environment';
 
 interface Props {
-  environments: string[];
+  environments?: string[];
 }
+
 export function EnvironmentBadge({ environments = [] }: Props) {
+  if (isEmpty(environments)) {
+    return <NotAvailableEnvironment />;
+  }
+
   return (
-    <ItemsBadge
-      items={environments ?? []}
-      multipleItemsMessage={i18n.translate('xpack.apm.servicesTable.environmentCount', {
-        values: { environmentCount: environments.length },
-        defaultMessage: '{environmentCount, plural, one {1 environment} other {# environments}}',
-      })}
-    />
+    environments &&
+    environments.length > 0 && (
+      <ItemsBadge
+        items={environments ?? []}
```

Comment:
I think this point `environment` can't be undefined
```suggestion
        items={environments}
```